softlayer/resource_softlayer_global_ip.go

```diff
@@ -45,6 +47,16 @@ func resourceSoftLayerGlobalIp() *schema.Resource {
 			"routes_to": &schema.Schema{
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					address := v.(string)
+					if net.ParseIP(address) == nil {
+						errors = append(errors, fmt.Errorf("Invalid IP"))
+					}
+					return
+				},
+				DiffSuppressFunc: func(k, o, n string, d *schema.ResourceData) bool {
+					return net.ParseIP(o).String() == net.ParseIP(n).String()
+				},
 			},
```

Comment:
`net.ParseIP()` will parse both compressed and uncompressed formats, and serialize (`.String()`) to the compressed format always.

https://play.golang.org/p/uASQqYtCpF